keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/ApplicationControllerTests.kt

```diff
@@ -44,6 +45,7 @@ import strikt.assertions.containsExactly
   webEnvironment = MOCK
 )
 @AutoConfigureMockMvc
+@DirtiesContext
```

Comment:
I don't understand how this can leak context when it's dropping all the repositories, unless there's a bug in one of the `dropAll` implementations

Reply:
`InMemoryDeliveryConfigRepository` doesn't drop content from its `applicationConstraintMapper` or `lastCheckTimes` properties for example

Reply:
and there are a bunch of things in `InMemoryArtifactRepository` that aren't cleared properly

Reply:
Yeah, I remember fixing one of those in a previous PR. I'll take a look, but it was worth understanding that context is indeed shared by default and how to force it not to be.

Reply:
It's worth knowing, but this annotation will have an impact on test execution time as it will require the Spring context to be re-created. I'd rather not get into the habit of using it.

Reply:
I agree. Having said that, I've done my best to clear all in-memory representations within the repositories and, upon removing the annotation, that particular test is still failing (whereas all tests pass on my machine). Any other ideas?

Reply:
BTW, just a thought: most test classes clear the in-memory repos in an `after` block that applies to all tests. Which means that, after the last test in that class runs, nothing does. What should be the pattern to make sure we clean up after finishing each set of tests?

Reply:
Actually, the failing test should be using SQL, not in-memory repos...

```
> Task :keel-sql:test

com.netflix.spinnaker.keel.tx.DeliveryConfigTransactionTests > minutests()[3][2] FAILED
    org.opentest4j.AssertionFailedError: ▼ Expect that true:
      ✗ is false
        found true
```

@asher any thoughts on why this might pass locally and fail here?

Reply:
This is nuts. It now passed without any related changes. 🤷‍♂

Reply:
Well, that's the nature of non-deterministic interference